example/helloworld/client.py

```diff
@@ -10,7 +10,33 @@
     channel = Channel(loop=asyncio.get_event_loop())
     stub = GreeterStub(channel)
 
-    print(await stub.SayHello(HelloRequest(name='World')))
+    # UNARY_UNARY - simple RPC
+    print(await stub.SayHello(HelloRequest(name='you')))
+    async with stub.SayHello.open() as stream:
+        await stream.send_message(HelloRequest(name='yall'))
+        reply = await stream.recv_message()
+        print(reply)
+
+    # UNARY_STREAM - response streaming RPC
+    print(await stub.SayHelloGoodbye(HelloRequest(name='yall')))
+    async with stub.SayHelloGoodbye.open() as stream:
+        await stream.send_message(HelloRequest(name='yall'), end=True)
+        replies = [reply async for reply in stream]
+        print(replies)
+
+    # STREAM_UNARY - request streaming RPC.
+    msgs = [HelloRequest(name='Rick'), HelloRequest(name='Morty')]
+    print(await stub.SayHelloToManyAtOnce(msgs))
+    async with stub.SayHelloToManyAtOnce.open() as stream:
+        for msg in msgs:
+            last_msg = msg == msgs[-1]
```

Comment:
This comparison will fail when you try to send some messages, identical to the last message in the list. Also such comparisons in loops are not the best option in terms of performance. Another possibility to end outgoing messages stream is to call `await stream.end()` method (will send one extra http/2 frame I think).

Reply:
Modified as you suggested.